### ISSUE TITLE
Commands do not work

### ISSUE THREAD

**Issue by Krunch** [SEModCommunity/SE-Community-Mod-API#90]
I already posted about this issue on the forum thread of this mod. I cannot get the commands to work at all. Everything else works fine tho, the gui, the auto-save seems all to work fine... but not the chat commands. There is also some path auto-detection issues that have to be fixed where if you have Steam installed but your Space Engineers server is not installed using Steam, it will pick the wrong path for the game.

Posts I made regarding this issue on the forums:
http://forums.keenswh.com/post/show_single_post?pid=1283727552&postcount=191
http://forums.keenswh.com/post/show_single_post?pid=1283765895&postcount=209

**Comment by DraygoKorvan** [comment]
Yeah we ran through everything as far as settings are concerned so, on to the fixing.

**Comment by chessmaster42** [comment]
Have you tried running with only the gamepath and instance set and none of the other command line args? Are you using any plugins? Have you tried the chat commands from the server GUI chat tab to see if it works there?

Please post both log files to pastebin.com and put the links here. Do NOT copy the log content into a comment.

**Comment by Krunch** [comment]
I will do this and post logs once the extender has been updated for the latest hotfix. Unfortunately my game client updated.

**Comment by chessmaster42** [comment]
Any update on this? Also can you confirm that your steam ID is on the list of administrators for your server config?

**Comment by Krunch** [comment]
I won't be able to test until tomorrow. My steam ID is in the administrators list, I added it through the regular Space Engineers launcher. When ingame, I can kick people and next to my name I see the stars (*****) which I guess means I am an administrator on the server.

**Comment by Krunch** [comment]
I tested as you said without the "autostart debug nogui" and the commands now work. Here is the pastebin logs:
http://pastebin.com/BpFGEn3P
http://pastebin.com/sJuKaj51

As you can see the only error there is is the GameInstallationInfoException error which in my case is normal like I explained in my previous forums posts (http://forums.keenswh.com/post/show_single_post?pid=1283765895&postcount=209), because it does not properly detect the gamepath (and doesn't even use the parameter I defined in the command line because of how the source is coded).

On the other hand now, when I put "autostart debug nogui" back in the command line arguments, the commands stop working. Here are the logs:
http://pastebin.com/u5zqF4V0
http://pastebin.com/4S7XzCZ1

I would assume there is a bug where if the "nogui" is defined the chat commands stop working. Also a fix for the gamepath detection would be appreciated, because honestly it is unnecessarily complicated at the moment. You could simply auto-detect using the location of the server extender executable. Also making the command line argument "gamepath" override the auto-detection would be more appropriate.

**Comment by Krunch** [comment]
Any news on this?

**Comment by chessmaster42** [comment]
The gamepath bug will be fixed in the next version so that if the command line arg is defined that will take precedence. As for the chat commands I still haven't been able to reproduce the problem but will continue working on it.

**Comment by chessmaster42** [comment]
The gamepath changes are complete so that should be working for you now. I haven't yet been able to reproduce the chat commands issue so please provide some more logs and steps to reproduce.

**Comment by Krunch** [comment]
I updated to latest version and the gamepath works. What is better is that commands also work now, so whatever you did with the WCF thingy fixed the commands too.

There is only a slight issue with spawning cargo ships, I tried spawning 3 times. The first 2 times it did nothing(messages showed in-game, but no ship appeared ingame), and the third time it crashed the server. Here are the logs:
http://pastebin.com/vMn2MXer
http://pastebin.com/uuqKPLRS
http://pastebin.com/t7Nqr69S
